test/anvil/entrypointsimulations.test.ts

```diff
@@ -0,0 +1,252 @@
+      const op = await fillAndSign({ sender: account.address, verificationGasLimit: 1000 }, accountOwner, entryPoint)
+      await expect(simulateValidation(op, entryPoint.address)).to
+        .revertedWith('AA23 reverted (or OOG)')
+    })
+
+    it('should succeed if validateUserOp succeeds', async () => {
+      const op = await fillAndSign({ sender: account1.address }, accountOwner1, entryPoint)
+      await fund(account1)
+      await simulateValidation(op, entryPoint.address)
+    })
+
+    it('should return empty context if no paymaster', async () => {
+      const op = await fillAndSign({ sender: account1.address, maxFeePerGas: 0 }, accountOwner1, entryPoint)
+      const { returnInfo } = await simulateValidation(op, entryPoint.address)
+      expect(returnInfo.paymasterContext).to.eql('0x')
+    })
+
+    it('should return stake of sender', async () => {
+      const stakeValue = BigNumber.from(123)
+      const unstakeDelay = 3
+      const { proxy: account2 } = await createAccount(ethersSigner, await ethersSigner.getAddress(), entryPoint.address)
+      await fund(account2)
+      await account2.execute(entryPoint.address, stakeValue, entryPoint.interface.encodeFunctionData('addStake', [unstakeDelay]))
+      const op = await fillAndSign({ sender: account2.address }, ethersSigner, entryPoint)
+      const result = await simulateValidation(op, entryPoint.address)
+      expect(result.senderInfo.stake).to.equal(stakeValue)
+      expect(result.senderInfo.unstakeDelaySec).to.equal(unstakeDelay)
+    })
+
+    it('should prevent overflows: fail if any numeric value is more than 120 bits', async () => {
+      const op = await fillAndSign({
+        preVerificationGas: BigNumber.from(2).pow(130),
+        sender: account1.address
+      }, accountOwner1, entryPoint)
+      await expect(
+        simulateValidation(op, entryPoint.address)
+      ).to.revertedWith('gas values overflow')
+    })
+
+    it('should fail creation for wrong sender', async () => {
+      const op1 = await fillAndSign({
+        initCode: getAccountInitCode(accountOwner1.address, simpleAccountFactory),
+        sender: '0x'.padEnd(42, '1'),
+        verificationGasLimit: 30e6
+      }, accountOwner1, entryPoint)
+      await expect(simulateValidation(op1, entryPoint.address))
+        .to.revertedWith('AA14 initCode must return sender')
+    })
+
+    it('should report failure on insufficient verificationGas (OOG) for creation', async () => {
+      const initCode = getAccountInitCode(accountOwner1.address, simpleAccountFactory)
+      const sender = await entryPoint.callStatic.getSenderAddress(initCode).catch(e => e.errorArgs.sender)
+      const op0 = await fillAndSign({
+        initCode,
+        sender,
+        verificationGasLimit: 5e5,
+        maxFeePerGas: 0
+      }, accountOwner1, entryPoint)
+      // must succeed with enough verification gas.
+      await simulateValidation(op0, entryPoint.address, { gas: '0xF4240' })
```

Comment:
why hex?

Reply:
The third parameter of `simulateValidation` is a transaction details object and is fed directly to the `eth_call` RPC method as `{...tx}`. It does not pass through the Ethers.js where `gasLimit?: string | number | BigNumber;` is translated into a hex `gas` field.